src/components/DomainSelector.vue

```diff
@@ -16,22 +16,31 @@
   </div>
 </template>
 
-<script>
-export default {
-  name: 'domain-selector',
-  props: {
-    domainsList: Array,
-    selectedDomain: String,
-  },
-  methods: {
-    isDomainSelected(name) {
-      return this.selectedDomain === name;
-    },
-    select(newDomain) {
-      this.$emit('selectedDomain', newDomain);
-    },
-  },
-};
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Prop } from 'vue-property-decorator'
+
+@Component({
+  name: 'domain-selector'
+})
+export default class DomainSelector extends Vue {
+  @Prop({
+    default: () => [],
+    type: Array
+  })
+  readonly domainsList!: Array<string>
```

Comment:
So there is two type declaration here:
- the one in the decorator @Prop is meant for vue, and will be enforced at _run time_
- the other directly after the prop name is meant for typescript, and will be enforced at _transpile time_

Reply:
:+1:

Reply:
:+1: